src/main/scala/Core/Top.scala

```diff
@@ -35,23 +37,25 @@ class VerilatorTop(cfg: FlexpretConfiguration) extends AbstractTop(cfg) {
     core.io.int_exts.foreach(_ := false.B)
 
     // Catch termination from core
-    when(core.io.host.to_host === "hdeaddead".U) {
-        printf(cf"SUCCESS: FlexPRET terminated execution\n")
-        io.stop := true.B
-    }
-
-    // Catch abort from core
-    when(core.io.host.to_host === "hdeadbeef".U) {
-        printf(cf"ERROR: FlexPRET aborted simulation\n")
-        assert(false.B, "Emulation stopped")
-    }
-
-    // Handle printfss
-    when(core.io.host.to_host === "hbaaabaaa".U) {
-      regPrintNext := true.B
-    }.elsewhen(regPrintNext) {
-      printf(cf"${core.io.host.to_host}\n")
-      regPrintNext := false.B
+    for (tid <- 0 until cfg.threads) {
+        when(core.io.host.to_host(tid) === "hdeaddead".U) {
+            io.stop := true.B
+        }
+
+        // Catch abort from core
+        when(core.io.host.to_host(tid) === "hdeadbeef".U) {
+            io.stop := true.B
+        }
+
+        io.to_host(tid) := core.io.host.to_host(tid)
+
+        // Handle printfss
+        when(core.io.host.to_host(tid) === "hbaaabaaa".U) {
+          regPrintNext(tid) := true.B
+        }.elsewhen(regPrintNext(tid)) {
+          printf(cf"[${tid}]: ${core.io.host.to_host(tid)}\n")
+          regPrintNext(tid) := false.B
```

Comment:
Yes, but this is for the _fp_print function, which I haven't removed yet. I suppose I didn't remove it since printf wasn't working fully, but it is now so I guess we can remove _fp_print entirely.

Reply:
I removed the _fp_print now, will look into removing the other to_host stuff from Top.scala as well.